CounterStrike2GSI/Nodes/Provider.cs

```diff
@@ -1,49 +1,56 @@
-﻿using Newtonsoft.Json.Linq;
-
-namespace CounterStrike2GSI.Nodes
-{
-    /// <summary>
-    /// Information about the provider of this GameState.
-    /// </summary>
-    public class Provider : Node
-    {
-        /// <summary>
-        /// Game name.
-        /// </summary>
-        public readonly string Name;
-
-        /// <summary>
-        /// Game Steam AppID.
-        /// </summary>
-        public readonly int AppID;
-
-        /// <summary>
-        /// Game version.
-        /// </summary>
-        public readonly int Version;
-
-        /// <summary>
-        /// Local player's Steam ID.
-        /// </summary>
-        public readonly string SteamID;
-
-        internal Provider(JObject parsed_data = null) : base(parsed_data)
-        {
-            Name = GetString("name");
-            AppID = GetInt("appid");
-            Version = GetInt("version");
-            SteamID = GetString("steamid");
-        }
-
-        /// <inheritdoc/>
-        public override string ToString()
-        {
-            return $"[" +
+﻿using Newtonsoft.Json.Linq;
+
+namespace CounterStrike2GSI.Nodes
+{
+    /// <summary>
+    /// Information about the provider of this GameState.
+    /// </summary>
+    public class Provider : Node
+    {
+        /// <summary>
+        /// Game name.
+        /// </summary>
+        public readonly string Name;
+
+        /// <summary>
+        /// Game Steam AppID.
+        /// </summary>
+        public readonly int AppID;
+
+        /// <summary>
+        /// Game version.
+        /// </summary>
+        public readonly int Version;
+
+        /// <summary>
+        /// Local player's Steam ID.
+        /// </summary>
+        public readonly string SteamID;
+
+        /// <summary>
+        ///  Timestamp of the GameState data.
+        /// </summary>
+        public readonly int Timestamp;
+
+        internal Provider(JObject parsed_data = null) : base(parsed_data)
+        {
+            Name = GetString("name");
+            AppID = GetInt("appid");
+            Version = GetInt("version");
+            SteamID = GetString("steamid");
+            Timestamp = GetLong("timestamp");

```

Comment:
Disassembly shows that the function used for populating `appid` and `version` fields is the same used for `timestamp`.

```suggestion
            Timestamp = GetInt("timestamp");
```